frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/TicfCriHandler.java

```diff
@@ -0,0 +1,98 @@
+
+import static java.lang.String.format;
+
+public class TicfCriHandler implements RequestHandler<TICFCRIRequest, Void> {
+
+    private final HttpClient httpClient;
+    private final ConfigurationService configurationService;
+    private final CloudwatchMetricsService cloudwatchMetricsService;
+
+    private static final SerializationService serialisationService =
+            SerializationService.getInstance();
+
+    public TicfCriHandler(
+            HttpClient httpClient,
+            ConfigurationService configurationService,
+            CloudwatchMetricsService cloudwatchMetricsService) {
+        this.httpClient = httpClient;
+        this.configurationService = configurationService;
+        this.cloudwatchMetricsService = cloudwatchMetricsService;
+    }
+
+    public TicfCriHandler(ConfigurationService configurationService) {
+        this.httpClient = HttpClient.newHttpClient();
+        this.configurationService = configurationService;
+        this.cloudwatchMetricsService = new CloudwatchMetricsService();
+    }
+
+    public TicfCriHandler() {
+        this(ConfigurationService.getInstance());
+    }
+
+    private static final org.apache.logging.log4j.Logger LOG =
+            LogManager.getLogger(TicfCriHandler.class);
+
+    @Override
+    public Void handleRequest(TICFCRIRequest input, Context context) {
+        LOG.info("received request to TICF CRI Handler");
+        var environmentForMetrics = Map.entry("Environment", configurationService.getEnvironment());
+        try {
+            var response = sendRequest(input);
+            var statusCode = String.valueOf(response.statusCode());
+            LOG.info("Response received from TICF CRI Service with status %s", statusCode);
+            cloudwatchMetricsService.incrementCounter(
+                    "TicfCriResponseReceived",
+                    Map.ofEntries(environmentForMetrics, Map.entry("StatusCode", statusCode)));
+        } catch (HttpTimeoutException e) {
+            var errorDescription =
+                    format(
+                            "Request to TICF CRI timed out with timeout set to %d",
+                            configurationService.getTicfCriServiceCallTimeout());
+            logAndSendMetricsForInterventionsError(errorDescription, "TicfCriServiceTimeout");
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            var errorDescription = format("Error occurred in the TICF CRI Handler: %s", e);
+            logAndSendMetricsForInterventionsError(errorDescription, "TicfCriServiceError");
+        } catch (IOException e) {
+            var errorDescription = format("Error occurred in the TICF CRI Handler: %s", e);
+            logAndSendMetricsForInterventionsError(errorDescription, "TicfCriServiceError");
+        }
+        return null;
+    }
+
+    private void logAndSendMetricsForInterventionsError(String errorDescription, String metric) {
+        LOG.warn(errorDescription);
```

Comment:
Why not do this at error level?

Reply:
So I did this because I didn't want this to trigger an alert as part of phase 1 since we're firing and forgetting, but maybe here I'll differentiate between an error that's on our side (IO exception or interrupted exception) and an error on the TICF side (timeout) and error on ones on our side.